bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v2/PageHelpers.java

```diff
@@ -0,0 +1,172 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2018 Adobe Systems Incorporated
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.internal.models.v2;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import javax.annotation.Nonnull;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.request.RequestParameter;
+import org.apache.sling.api.wrappers.SlingHttpServletRequestWrapper;
+
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.components.ComponentContext;
+import com.day.cq.wcm.api.designer.Style;
+
+final class PageHelpers {
```

Comment:
This seems unnecessary as it does not look they could be reused outside the `Page` implementation. I would move the helper methods as `protected` members of `PageImpl`.

Reply:
I've put them aside as many of those are very loosely related to the Page Object. This way I had them as static functions that were very easy to unit-test.

I fully understand your design decision and I'm just merging them to the v2.PageImpl Object. Would you prefer to keep them static (and keep the unit-tests) or should I also change them to Instance Methods and have them tested just as a by-product (as implementation details)?

Reply:
How about a `v3.PageImpl` since we're bumping the component version anyway.

I'd try and keep the units tests.